src/hotspot/cpu/aarch64/macroAssembler_aarch64.hpp

```diff
@@ -581,6 +581,14 @@ class MacroAssembler: public Assembler {
     mrs(0b011, 0b0000, 0b0000, 0b001, reg);
   }
 
+  inline void get_nzcv(Register reg) {
+    mrs(0b011, 0b0100, 0b0010, 0b000, reg);
+  }
+
+  inline void set_nzcv(Register reg) {
+    msr(0b011, 0b0100, 0b0010, 0b000, reg);
+  }
+
```

Comment:
We've tried to to add such things speculatively, as much as anything else because it's impossible to test.

Reply:
Could @fisk provide an example of where this might be used?
My assumption is that you are looking to execute code verification within, say, a GC memory barrier, and the code has been inserted in between where the condition codes might have been set and then used. This could avoid breaking the existing code when the verification code itself sets the condition codes.
It's not all of the pstate, but then this is just userspace, so it ought to be ok?

Reply:
> Could @fisk provide an example of where this might be used?
> 
> My assumption is that you are looking to execute code verification within, say, a GC memory barrier, and the code has been inserted in between where the condition codes might have been set and then used. This could avoid breaking the existing code when the verification code itself sets the condition codes.
> 
> It's not all of the pstate, but then this is just userspace, so it ought to be ok?
> 
> 

Your guess is spot on. We have a bunch of GC barrier nodes in C1, and C1 sprinkles oop verification all over the place - including when spilling oops between lir ops - sometimes between producing a condition code and consuming it. So I figured the verification code shouldn't really alter the state of execution such that it does something different. And I don't really want to reduce the amount of verification, because it was very useful in tracking down bugs.

Reply:
OK, I'm not wild about checking in dead code, but if it's imminently needed, OK. Please add the instructions to  `test/hotspot/gtest/aarch64/aarch64-asmtest.py` so they get tested..

Reply:
The bits for the instructions are correct. I presume there'll be a train of patches for ZGC generational to necessitate it.

Reply:
I suggest adding one comment here. How about:
```
// macro instructions for accessing and updating 
// PSTATE.{N,Z,C,V} fields at EL0
//
// NZCV:  op1 = 011
//        CRn = 0100
//        CRm = 0010
//        op2 = 000
```

Reply:
As a general rule, we don't add commentary that duplicates information in the Architecture Reference Manual. In this case, it's obvious that `MRS <Xt>, NZCV` is intended, and the information above is on Page C5-689 of ARM DDI 0487H.a

Reply:
> OK, I'm not wild about checking in dead code, but if it's imminently needed, OK. Please add the instructions to `test/hotspot/gtest/aarch64/aarch64-asmtest.py` so they get tested..

I added mrs and msr tests for the system registers we use to aarch64-asmtest.py